### PR TITLE
Delete Database, Refactor Lyric search logic

### PR DESCRIPTION
**Pull request by brandonxia01**
Some browsers don't support web sql database, which was a big problem. When that was the case we had to fallback to an in-code algorithm to search for songs. This PR basically removes the database altogether and makes every search use an in-house algorithm for song matching. 

This PR accomplishes a few things:
* Deletes the sqlite/websql databases from the app (rip me)
* Lyrics will be `fetch`ed on every instance the app is loaded up (ie: on refresh), which will give users updates immediately when we fix lyrics in the `Resources` repo.
* Extract the song/lyric search logic into a `SongsService` file. Future enhancements to how we do lyric lookup should go there.
* Cleans up the console logs by using `console.debug` instead. They are still visible by turning on `verbose` log level in the browser.

The main regression for this is that we won't be able to search by lyrics until we enhance the song matching algorithm. We will still get fairly accurate results though when searching by title or author.

```
this PR has pros:
- Removes database from app (Same behavior regardless of browser)
- Un-caches the lyric file so they get refreshed when you refresh the page (no stale lyric fixes)
- Refactored a home for the in-house song matching algorithm.

and cons:
- Temporary loss of being able to search by full-text lyrics (until the next PR when I or someone else can enhance the search algorithm)
```

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -22,7 +22,6 @@ import "./theme/variables.css";
 import HomePage from "./pages/HomePage";
 import BookPage from "./pages/BookPage";
 import SongPage from "./pages/SongPage";
-import { DbManager } from "./database/DbManager";
```
Comment (brotatotes) [inline src/App.tsx:25]: goodbye fren 😢

File: src/test/App.test.tsx [hunk 2]
```diff
@@ -69,7 +69,7 @@ describe("App", () => {
   }, 10000);
 
   it("searching terms not found displays no results", async () => {
-    await verifySearchResults(page, "gibberish", []);
```
Comment (brotatotes) [inline src/test/App.test.tsx:72]: was this no longer passing? why would that be?

File: src/utils/SongUtils.tsx [hunk 3]
```diff
@@ -13,6 +13,8 @@ export const shlJsonUrl =
 
 export const SHL_BOOK_ID = "shl";
 
+const emptyMap = new Map();
```
Comment (brotatotes) [inline src/utils/SongUtils.tsx:16]: nit: If this isn't being used anywhere else, let's just inline it in `PLACEHOLDER_SONG`

**Review by brotatotes — APPROVED** [review]
minor comments only. Approved.

**Comment by brotatotes** [comment]
Thanks for taking the time to clean a lot of things up in the codebase 👍

**Comment by brotatotes** [comment]
Sad to see your DB go. At least we can try to take this as a learning experience. I suppose this is why DB/backend isusually separate from frontend. Thank you for your service, DB.
image

File: src/utils/SongUtils.tsx (hunk 3 above)
Comment (brandonxia01) [reply src/utils/SongUtils.tsx:16]: Done 👍

File: src/test/App.test.tsx (hunk 2 above)
Comment (brandonxia01) [reply src/test/App.test.tsx:72]: Yeah, just the way the search algorithm is now is matching by string similarity, not pattern matching. This was matching the author's name "Gibbs". Definitely the match algorithm can be refined but it works ok in most cases.

Comment (brotatotes) [reply src/test/App.test.tsx:72]: I see! IMO this is actually an improvement as well. Very nice.

**Review by brotatotes — APPROVED** [review]

**Pull request merged**